src/tcgwars/logic/impl/gen3/RubySapphireNG.groovy

```diff
@@ -450,14 +541,21 @@ public enum RubySapphireNG implements LogicCardInfo {
         pokePower "Water Call", {
           text "Once during your turn (before your attack), you may attach a [W] Energy card from your hand to your Active Pokémon. This power can't be used if Swampert is affected by a Special Condition."
           actionA {
+            checkNoSPC()
+            checkLastTurn()
+            assert my.hand.filterByType(BASIC_ENERGY).filterByEnergyType(W) : "You have no [W] Energy cards in your hand"
```

Comment:
Theoretically this PokePower could allow a Special Energy card that provides W energy while not attached couldn't it? Should this force filtering to `BASIC_ENERGY`? `filterByEnergyType` should check the types provided to the Special Energy in its implementation which should all now be correct (empty for most Special Energy cards)

Reply:
```suggestion
            assert my.hand.filterByEnergyType(W) : "You have no [W] Energy cards in your hand"
```